llvm/lib/Transforms/Utils/Local.cpp

```diff
@@ -2779,6 +2779,11 @@ llvm::removeAllNonTerminatorAndEHPadInstructions(BasicBlock *BB) {
   Instruction *EndInst = BB->getTerminator(); // Last not to be deleted.
   // RemoveDIs: erasing debug-info must be done manually.
   EndInst->dropDbgValues();
+
+  if (isa<ReturnInst>(EndInst) && EndInst->getNumOperands() > 0 &&
+      isa<Instruction>(EndInst->getOperand(0)))
+    EndInst->setOperand(0, PoisonValue::get(EndInst->getOperand(0)->getType()));
```

Comment:
We shouldn't hardcode this to `ReturnInst`. We should replace any operand in the terminator. E.g. if this is a `br`, we should also set the condition to poison. (Skip the replacement for token types though.)

You should also do a similar change in `InstCombinerImpl::handleUnreachableFrom`.

Reply:
@nikic got it. 
//‘ret’, ‘br’, 
// ‘switch’, ‘indirectbr’, ‘invoke’, ‘callbr’ ‘resume’, 
//‘catchswitch’, ‘catchret’, ‘cleanupret’, and ‘unreachable’.

I'll implement codes to handle above terminators.
except `CatchSwitch`. it is element of EHPad can be skipped.

Reply:
The handling should be generic, not for specific terminator types.